applications/nucleo_io/targets/freertos.armv6m.st-stm32f091rc-nucleo-dev-board/ServoConsumer.hxx

```diff
@@ -0,0 +1,135 @@
+#ifndef _APPLICATIONS_IO_BOARD_TARGET_SERVOCONSUMER_HXX_
+#define _APPLICATIONS_IO_BOARD_TARGET_SERVOCONSUMER_HXX_
+
+#include "config.hxx"
+#include "os/MmapGpio.hxx"
+
+const unsigned servo_ticks_0 = configCPU_CLOCK_HZ * 1 / 1000;   // 48k
+const unsigned servo_ticks_180 = configCPU_CLOCK_HZ * 2 / 1000; // 96k
+
+uint8_t dummy_var[1] = {0x0};
+constexpr const MmapGpio8 DUMMY_GPIO(
+    dummy_var, /*offset=*/0, /*is_output=*/false);
+
+// GPIOBit has a <template HW> ctor that won't work if
+// supplied Gpio* is a PWMGPO; no idea why.
+class MyGPIOBit : public openlcb::BitEventInterface
+{
+public:
+    MyGPIOBit(openlcb::Node *node, openlcb::EventId event_on,
+        openlcb::EventId event_off, const Gpio *gpio)
+        : BitEventInterface(event_on, event_off)
+        , node_(node)
+        , gpio_(gpio)
+    {
+    }
+
+    openlcb::EventState get_current_state() override
+    {
+        return gpio_->is_set() ? openlcb::EventState::VALID
+                               : openlcb::EventState::INVALID;
+    }
+
+    void set_state(bool new_value) override
+    {
+        gpio_->write(new_value);
+    }
+
+    openlcb::Node *node() override
+    {
+        return node_;
+    }
+
+private:
+    openlcb::Node *node_;
+    const Gpio *gpio_;
+};
+
+// Basically a specialized ConfiguredConsumer.
+// Can't subclass ConfiguredConsumer here because ServoConsumerConfig
+// isn't a subclass of ConsumerConfig,
+class ServoConsumer : public DefaultConfigUpdateListener
+{
+public:
+    ServoConsumer(
+        openlcb::Node *node, const openlcb::ServoConsumerConfig &cfg, PWM *pwm)
+        : DefaultConfigUpdateListener()
+        , pwm_(pwm) // save for apply_config, where we actually use it.
+        , pwm_gpo_(nullptr) // not initialized until apply_config
+        , gpio_impl_(node, 0, 0, &DUMMY_GPIO)
+        , consumer_(&gpio_impl_) // don't connect consumer to PWM yet
+        , cfg_(cfg)
+    {
+    }
+
+    UpdateAction apply_configuration(
+        int fd, bool initial_load, BarrierNotifiable *done) OVERRIDE
+    {
+        AutoNotify n(done);
+
+        const openlcb::EventId cfg_event_min = cfg_.event_rotate_min().read(fd);
+        const openlcb::EventId cfg_event_max = cfg_.event_rotate_max().read(fd);
+        const uint8_t cfg_servo_min_pct = cfg_.servo_min_percent().read(fd);
+        const uint8_t cfg_servo_max_pct = cfg_.servo_max_percent().read(fd);
+
+        const unsigned servo_range_ticks = servo_ticks_180 - servo_ticks_0;
+
+        const unsigned cfg_srv_ticks_min =
+            servo_ticks_0 + (servo_range_ticks * (cfg_servo_min_pct / 100.0));
```

Comment:
This is going to link in floating point libraries which is somewhat undesirable.
Please check if you can get equivalent results using Fixed16 in src/utils/Fixed16.hxx. It also supports division.

Another alternative is to rewrite this entire expression to a numerically more stable weighted average:
(min_pct * ticks_0 + (100-min_pct) * ticks_180) / 100

Reply:
Done. I used this strategy, but I switched `min_pct` with `100-min_pct`. Otherwise, what's calculated is the ticks at `100-percent` .